### PR TITLE
ROX-21221: Default policy updates and migration

### PR DESCRIPTION
**Pull request by theencee**
## Description

This PR combines multiple different policy updates into one PR so that migrations can also be handled in just one step.

It cherry-picks from the following PRs:
https://github.com/stackrox/stackrox/pull/9294
https://github.com/stackrox/stackrox/pull/9172
https://github.com/stackrox/stackrox/pull/8759

And also contains policy exclusions from ROX-21221.

There are a lot of files changes, but a large amount of it is boiler plate. It is recommended to review this PR commit by commit so that it's not overwhelming.

Here are what each file corresponds to:

All files in `pkg/defaults/policies/files` -> This is the actual policy and the updates to each policy. Please review this carefully.

All migration code is in `migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates`

Within the migration folder, the folder `policies_before_and_after` contains two folders: `before` and `after`. The migration diffs the policy in before with the ones in after to determine how the policy should be changed. Thus the files in `before` are what the policy looked like before this PR, and `after` is what it should be after migration.

`CHANGELOG` -> only changes that changes names or policy criteria is listed

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] Evaluated and added CHANGELOG entry if required
- [x] Determined and documented upgrade steps
- [x] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Unit tests
Manual test:
 - [x] New install should show policy updated
 - [ ] Updates should also show policy updated 

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/defaults/policies/files/sensitive_files.json [hunk 1]
```diff
@@ -267,6 +267,23 @@
...
+    {
+      "name": "Don't alert on openshift-logging namespace",
+      "deployment": {
+        "scope": {
+          "namespace": "openshift-logging"
+        }
+      }
+    },
+    {
+      "name": "Don't alert on deployment  in namespace openshift-cloud-controller-manager",
```
Comment (rukletsov) [inline pkg/defaults/policies/files/sensitive_files.json:280]:
```suggestion
      "name": "Don't alert on deployment aws-cloud-controller-manager in namespace openshift-cloud-controller-manager",
```

File: migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/policies_before_and_after/after/sensitive_files.json [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    {
+      "name": "Don't alert on openshift-logging namespace",
+      "deployment": {
+        "scope": {
+          "namespace": "openshift-logging"
+        }
+      }
+    },
+    {
+      "name": "Don't alert on deployment  in namespace openshift-cloud-controller-manager",
```
Comment (rukletsov) [inline migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/policies_before_and_after/after/sensitive_files.json:280]:
```suggestion
      "name": "Don't alert on deployment aws-cloud-controller-manager in namespace openshift-cloud-controller-manager",
```

File: migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+var (
+
+	//go:embed policies_before_and_after
+	policyDiffFS embed.FS
+
+	// Update only if the existing name, description and policy sections haven't changed.
+	fieldsToCompare = []policymigrationhelper.FieldComparator{
+		policymigrationhelper.NameComparator,
+		policymigrationhelper.RationaleComparator,
```
[the remark this answers is not in the document]
Comment (theencee) [reply migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go:23]: Copy paste from previous migration. I'll remove

File: migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+func migrate(database *types.Databases) error {
+
+	db := database.GormDB
+	return policymigrationhelper.MigratePoliciesWithDiffsAndStoreV2(
+		policyDiffFS,
+		policyDiffs,
+		// Get policy with specified id
+		func(ctx context.Context, id string) (*storage.Policy, bool, error) {
+			var foundPolicy schema.Policies
+			result := db.WithContext(ctx).Table(schema.PoliciesTableName).Where(&schema.Policies{ID: id}).First(&foundPolicy)
```
Comment (ksurabhi91) [inline migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go:73]: can we add this as a generic function in migration helper?

File: migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+			if err != nil {
+				return nil, false, err
+			}
+			return storagePolicy, true, nil
+		},
+		// Upsert policy. Technically it should be just an update and not create because in theory policy has been verified to exist
+		func(ctx context.Context, policy *storage.Policy) error {
+			dbPolicy, err := schema.ConvertPolicyFromProto(policy)
+			if err != nil {
+				return err
```
[the remark this answers is not in the document]
Comment (theencee) [reply migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go:90]: Unfortunately no. The `schema` package there is specific to each migration. It's the frozen schema at the point of the migration. By moving it to the helper, we wouldn't be able to freeze the schema every migration.

File: migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go (hunk 4 above)
Comment (theencee) [reply migrator/migrations/m_198_to_m_199_policy_description_and_criteria_updates/migration_impl.go:73]: Unfortunately no. The `schema` package there is specific to each migration. It's the frozen schema at the point of the migration. By moving it to the helper, we wouldn't be able to freeze the schema every migration.

**Review by rukletsov — APPROVED** [review]

**Comment by rukletsov** [comment]
🚢 . Assuming scanner v4 tests can be ignored for now.

**Comment by theencee** [comment]
There were a few fixes for the scanner tests so let me rebase and try to see if it fixes it. But it should be unrelated to this PR

**Pull request merged**